contrib/parseq-lambda-names/src/main/java/com/linkedin/parseq/lambda/FindMethodCallAnalyzer.java

```diff
@@ -0,0 +1,179 @@
+
+    FindMethodCallMethodVisitor(int api, int access, String name, String desc, String signature, String[] exceptions) {
+      super(api, access, name, desc, signature, exceptions);
+    }
+
+    @Override
+    public void visitEnd() {
+      try {
+        Analyzer analyzer = new Analyzer(new SourceInterpreter());
+        Frame[] frames = analyzer.analyze(_classToAnalyze, this);
+
+        LabelNode label = findLineLabel(this.instructions, _lineNumberOfMethodCall);
+        int index = findMethodCall(this.instructions, label);
+
+        if (index != -1) {
+          List<String> localVariables = new ArrayList<>();
+
+          String field = "";
+
+          Frame f = frames[index];
+          boolean parsedThisOnce = false;
+          for (int j = 0; j < f.getStackSize(); ++j) {
+            SourceValue stack = (SourceValue) f.getStack(j);
+
+            Object insn = stack.insns.iterator().next();
+            if (insn instanceof VarInsnNode) {
+              VarInsnNode vinsn = (VarInsnNode) insn;
+              if (vinsn.var < this.localVariables.size()) {
+                String variable = ((LocalVariableNode) this.localVariables.get(vinsn.var)).name;
+
+                //This part is tricky: discard the first this.
+                if (variable.equals("this") && parsedThisOnce == false) {
+                  parsedThisOnce = true;
+                } else {
+                  localVariables.add(variable);
+                }
+              }
+            } else if (insn instanceof FieldInsnNode) {
+              FieldInsnNode fieldInstr = (FieldInsnNode) insn;
+              field = fieldInstr.name;
+            } else if (insn instanceof TypeInsnNode) {
+              TypeInsnNode typeInsnNode = (TypeInsnNode) insn;
+              if (typeInsnNode.getOpcode() == Opcodes.NEW) {
+                field = "new " + Util.extractSimpleName(typeInsnNode.desc, "/") + "()";
+              }
+            } else if (insn instanceof MethodInsnNode) {
+              MethodInsnNode methodInstr = (MethodInsnNode) insn;
+              if (methodInstr.getOpcode() == Opcodes.INVOKESPECIAL && methodInstr.name.equals("<init>")) {
+                field = "new " + Util.extractSimpleName(methodInstr.owner, "/") + "()";
+              } else {
+                Type methodType = Type.getMethodType(methodInstr.desc);
+                int retSize = methodType.getArgumentsAndReturnSizes() & 0x03;
+                if (retSize > 0) {
+                  field = methodInstr.name + Util.getArgumentsInformation(methodInstr.desc);
+                }
+              }
+            }
+          }
+
+          _inferredOperation = getInferredOperation(localVariables, field);
+        }
+      } catch (AnalyzerException e) {
+        //TODO: throw an exception that couldn't infer operation
+        LOGGER.error("Unable to analyze class, could not infer operation", e.getMessage());
```

Comment:
We probably should log on DEBUG level because better lambda names are an optional improvement and we don't want to fill log with this kind of error messages if e.g. someone uses unsupported JVM.